tests/test_tokenizer.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+import os
+import codecs
+
+from six import StringIO
+import pytest
+
+from segments.tokenizer import Tokenizer, Profile, GRAPHEME_COL
+from segments.tree import printMultigraphs
+
+
+def _test_path(fname):
+    return os.path.join(os.path.dirname(__file__), 'data', fname)
+
+
+def _test_data(fname):
+    with codecs.open(_test_path(fname), "r", "utf-8") as fp:
+        return fp.read()
+
+
+@pytest.fixture
+def tokenizer_with_profile():
+    return Tokenizer(_test_path('test.prf'))
+
+
+@pytest.fixture
+def tokenizer():
+    return Tokenizer()
+
+
+@pytest.mark.parametrize(
+    "lang", ['Kabiye', 'Brazilian_Portuguese', 'Vietnamese', 'Zurich_German'])
+def test_jipa(lang):
+    tokenize = Tokenizer()
+    assert tokenize(
+        _test_data(lang + '_input.txt'), ipa=True) == _test_data(lang + '_output.txt')
+
+
+def test_missing_header():
+    with pytest.raises(ValueError):
+        Profile({})
+
+
+def test_duplicate_grapheme(mocker):
+    mock_log = mocker.Mock()
+    mocker.patch('segments.tokenizer.logging', mocker.Mock(getLogger=lambda n: mock_log))
+    Profile({'Grapheme': 'a'}, {'Grapheme': 'a'})
+    assert mock_log.warn.call_args[0][0].startswith('line 3')
```

Comment:
can't we to
```python
logger = mocker.patch('segments.tokenizer.logging')
assert logger.getLogger.return_value.call_args...
```
also, can't we use https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_with
or even https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_once_with

Reply:
hm. Didn't get this to work.

Reply:
```python
def test_duplicate_grapheme(mocker):
    logging = mocker.patch('segments.tokenizer.logging')
    Profile({'Grapheme': 'a'}, {'Grapheme': 'a'})
    assert logging.getLogger.return_value.warn.call_args[0][0].startswith('line 3')
```